dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -747,6 +748,35 @@ public User updateLoggedInUserMetadata(@ApiParam(hidden = true)@Parameter(hidden
         return dbuser;
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @RolesAllowed({"admin", "curator"})
+    @Path("/user/{userId}/privileges")
+    @Operation(operationId = "addUserPrivileges", description = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class)
+    public User setUserPrivilege(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User authUser,
+                                 @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userID,
+                                 @ApiParam(value = "Set privilege for a user", required = true) PrivilegeRequest privilegeRequest) {
+        User user = userDAO.findById(userID);
+        if (user == null) {
+            throw new CustomWebApplicationException("User not found", HttpStatus.SC_NOT_FOUND);
+        }
+
+        if (authUser.getId() == user.getId()) {
+            throw new CustomWebApplicationException("You cannot modify your own privileges", HttpStatus.SC_FORBIDDEN);
+        }
+
+        if (privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin()) {
```

Comment:
Should add a comment on both clauses, conditions are better but still not arguably simply readable

Reply:
I might be overlooking something, but is `(privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin())`  simpler as `(privilegeRequest.isAdmin()  && !authUser.getIsAdmin())`

Reply:
I think the original logic is that an error should be thrown if the user admin status is different from the privilege request and that the authuser is not an admin. That way a curator can set the curator status only for an admin and not get blocked by it. Also, with `(privilegeRequest.isAdmin() && !authUser.getIsAdmin())`, the curator would be able to remove adminship since privilegeRequest.isAdmin() would be false and the error would not be thrown.

Reply:
Added a comment on both clauses.

Reply:
Ok, thanks for the explanation. I think I'm not reading the curator logic properly here. 
This code might be more readable with one branch for curator logic and one branch for admin logic.

Reply:
There isn't really an admin logic branch. It's more of the first branch checking for curator status and creating restrictions and the second branch setting the privileges since no errors were thrown. I have created two if statements instead of doing an if - else if and added additional comments. Hopefully that will make the logic more clear to read.